From 216d25722004094ba588a6aea3929756e7a3bb3d Mon Sep 17 00:00:00 2001
From: "i.Dark_Templar" <darktemplar@dark-templar-archives.net>
Date: Fri, 17 Dec 2021 23:48:40 +0300
Subject: [PATCH] Fix hidden NTFS mountpoints when /etc/mtab is a regular file

if /etc/mtab is a regular file,
/etc/mtab is used by defaul instead of /proc/self/mountinfo file,
and function mnt_fs_get_devno returned 0.
This led to NTFS mountpoints being hidden.
---
 src/core/kmountpoint.cpp | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/src/core/kmountpoint.cpp b/src/core/kmountpoint.cpp
index f41cff5ee..6c2ab0036 100644
--- a/src/core/kmountpoint.cpp
+++ b/src/core/kmountpoint.cpp
@@ -343,8 +343,11 @@ KMountPoint::List KMountPoint::currentMountPoints(DetailsNeededFlags infoNeeded)
 
 #elif HAVE_LIB_MOUNT
     if (struct libmnt_table *table = mnt_new_table()) {
-        // By default, parses "/proc/self/mountinfo"
-        if (mnt_table_parse_mtab(table, nullptr) == 0) {
+        // if "/etc/mtab" is a regular file,
+        // "/etc/mtab" is used by default instead of "/proc/self/mountinfo" file in function mnt_fs_get_devno,
+        // and returned dev is 0.
+        // This leads to NTFS mountpoints being hidden.
+        if (mnt_table_parse_mtab(table, "/proc/self/mountinfo") == 0) {
             struct libmnt_iter *itr = mnt_new_iter(MNT_ITER_FORWARD);
             struct libmnt_fs *fs;
 
-- 
2.32.0

